src/NITTA/Intermediate/Functions.hs

```diff
@@ -442,3 +468,130 @@ instance Var v => Locks (BrokenBuffer v x) v where
     locks = inputsLockOutputs
 instance Var v => FunctionSimulation (BrokenBuffer v x) v x where
     simulate cntx (BrokenBuffer (I a) (O vs)) = [(v, cntx `getCntx` a) | v <- elems vs]
+data LogicFunction v x
+    = LogicAnd (I v) (I v) (O v)
+    | LogicOr (I v) (I v) (O v)
+    | LogicNot (I v) (O v)
+    | LogicCompare Op (I v) (I v) (O v)
```

Comment:
Why it is LogicCompare? Why not a separated function or functions? Static types is better that runtime value cast.